lib/iris/tests/test_concatenate.py

```diff
@@ -217,15 +231,18 @@ def concatenate(cubes, order=None):
     result = cubelist.concatenate()
 
     for cube in result:
+        # Setting the cube.data clears the cube.dtype and cube.fill_value.
+        # We want to maintain the fill_value for testing purposes, even
+        # though we have lost the lazy data in order to pin down the array
+        # data order to overcome testing on different architectures.
```

Comment:
I've tried to leave some clue's (for my future self) but to be honest if we have a step change in the "persisting fill_value across a data setter operation" then we can always find the usage of `fill_value` in the code base.

I just added a bit more of context for concatenate because it's testing is a bit weird and, at the time, I thought an exception to the norm.